app/src/Grout.js

```diff
@@ -0,0 +1,273 @@
+const axios = require('axios');
+const qs = require('qs');
+
+
+class AbstractApiClass {
+    /*
+     * All methods that interact with Grout share a set of common queries, and
+     * need to be configured to talk to the API server. This base class provides
+     * common methods and configuration options for those classes.
+     *
+     * @class
+     */
+    constructor(url) {
+        /*
+         * The constructor function for AbstractApiClass.
+         *
+         * @params {string} url - The base URL for the Grout API endpoint. For
+         *                        an app running on the same server as the API,
+         *                        this might be e.g. 'localhost:8000/api'.
+         */
+        this.baseUrl = url;
+        this.url = this.baseUrl;
+    }
+
+    get(uuid) {
+        /*
+         * Retrieve an instance of a data type from the Grout API based on its
+         * UUID.
+         *
+         * @param {string} uuid - A unique identifier for the data type instance.
+         * @returns {object} - The JSON blob corresponding to the instance.
+         */
+        return new Promise((resolve, reject) => {
+            axios.get(`${this.url}${uuid}`)
+                .then(response => {
+                    resolve(response.data);
+                })
+                .catch(error => {
+                    reject(error);
+                })
+        });
+    }
+
+    create() {
+        throw new Error(`
+            Children of AbstractApiClass must implement the 'create' method.
+        `);
+    }
+
+    query () {
+        throw new Error(`
+            Children of AbstractApiClass must implement the 'query' method.
+        `);
+    }
+
+    all() {
+        /*
+         * Retrieve all instances of a given data type from the Grout API.
+         *
+         * @returns {Promise} - If successful, the promise contains an array of
+         *                      all data type instances, represented by JSON blobs.
+         *                      If unsuccessful, the promise throws an error.
+         */
+        return new Promise((resolve, reject) => {
+            axios.get(`${this.url}?limit=all`)
+                .then(response => {
+                    resolve(response.data.results);
+                })
+                .catch(error => {
+                    reject(error);
+                });
+        });
+    }
+
+}
+
+
+class Schema extends AbstractApiClass {
```

Comment:
I'm sympathetic to that, but my gut feeling is that `Schema` and `Type` are such general terms that this might become confusing or make things hard to grep for down the road; perhaps there's a shorter prefix we could use, like `Grout`?

Reply:
That's an important point! I'm open to whatever name is the most readable from the end user's perspective.

One thing I probably should have mentioned is that these class names are internal; the exports for the module expose them as attributes on the global `Grout` instance, e.g.:

```js
import Grout from 'Grout';

// Configure a Grout instance
grout = Grout('connection-string');

// Access method for `Schema.query`
grout.schemas.query(...);

// Access method for `Type.all`
grout.types.all();
```

Does that seem like a reasonable method of "prefixing" the class names?

Reply:
Yes, that's a good point! Unless someone is developing on the Grout JS API itself, they shouldn't ever have to interact with these types directly. So I think it's fine as-is. Sorry, I noticed this as I was reviewing but forgot to go back and revise this comment.